sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/JsonExpressionsSuite.scala

```diff
@@ -853,4 +853,26 @@ class JsonExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper with
         checkEvaluation(JsonObjectKeys(Literal(input)), expected)
     }
   }
+
+  test("is_json") {
+    Seq(
+      // valid JSON strings
+      ("", true),
+      ("{}", true),
+      ("[]", true),
+      ("[1, 2, 3]", true),
+      ("""{"key": 1}""", true),
+      ("""[1, 2, {"key" : "valid JSON array"}]""", true),
+      ("""{"k1": [1, 2, {"key": 5}], "k2": {"key2": [1, 2]}}""", true),
+      // invalid JSON strings
```

Comment:
Can you check nested invalid JSON objects?

Reply:
Test case added for nested invalid JSON objects.